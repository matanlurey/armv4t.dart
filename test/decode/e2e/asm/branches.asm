branches:
        ; Tests for branch operations

t050:
        ; ARM 1: Branch with exchange
        mov     r12, 50
        adr     r0, t051 + 1
        bx      r0

code16
align 2
t051:
        ; THUMB 5: Branch with exchange
        mov     r0, 51
        mov     r12, r0
        adr     r0, t052
        bx      r0

code32
align 4
t052:
        ; ARM 1: Branch without exchange
        mov     r12, 52
        adr     r0, t053
        bx      r0

t053:
        ; ARM 2: Branch forward
        mov     r12, 53
        b       t054

t055:
        ; ARM 2: Branch forward
        mov     r12, 55
        b       t056

t054:
        ; ARM 2: Branch backward
        mov     r12, 54
        b       t055

t057:
        ; ARM 2: Test link
        mov     r12, 57
        mov     pc, lr

t056:
        ; ARM 2: Branch with link
        mov     r12, 56
        bl      t057

branches_passed:
        mov     r12, 0
