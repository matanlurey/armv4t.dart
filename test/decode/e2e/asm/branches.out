; Automatically generated

mov r12, 50                     ; 0xe3a0c032
add r0, pc, 1                   ; 0xe28f0001
bx r0                           ; 0xe12fff10
???? <UnimplementedError>       ; 0x46842033
strmi r10, [r0, -r0]            ; 0x4700a000
mov r12, 52                     ; 0xe3a0c034
add r0, pc, 0                   ; 0xe28f0000
bx r0                           ; 0xe12fff10
mov r12, 53                     ; 0xe3a0c035
b 1                             ; 0xea000001
mov r12, 55                     ; 0xe3a0c037
b 3                             ; 0xea000003
mov r12, 54                     ; 0xe3a0c036
b 16777211                      ; 0xeafffffb
mov r12, 57                     ; 0xe3a0c039
mov pc, r14                     ; 0xe1a0f00e
mov r12, 56                     ; 0xe3a0c038
bl 16777211                     ; 0xebfffffb
mov r12, 0                      ; 0xe3a0c000